docs/learn/smart-contract-developers/create-nft-collection-with-lsp7-tokenId.md

```diff
@@ -0,0 +1,956 @@
+        bytes32 dataKey,
+        bytes memory dataValue
+    ) internal override {
+        require(
+            dataKey != _LSP8_REFERENCE_CONTRACT,
+            "LSP8ReferenceContractNotEditable"
+        );
+        LSP4DigitalAssetMetadata._setData(dataKey, dataValue);
+    }
+}
+```
+
+### Final LSP7 token contract
+
+<details>
+<summary>Click to expand/collapse the script</summary>
+
+```typescript
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.20;
+
+import {LSP7DigitalAsset} from "@lukso/lsp-smart-contracts/contracts/LSP7DigitalAsset/LSP7DigitalAsset.sol";
+import {LSP4DigitalAssetMetadata} from "@lukso/lsp-smart-contracts/contracts/LSP4DigitalAssetMetadata/LSP4DigitalAssetMetadata.sol";
+import {_LSP4_SUPPORTED_STANDARDS_KEY, _LSP4_SUPPORTED_STANDARDS_VALUE, _LSP4_METADATA_KEY} from "@lukso/lsp-smart-contracts/contracts/LSP4DigitalAssetMetadata/LSP4Constants.sol";
+import {_LSP8_REFERENCE_CONTRACT} from "@lukso/lsp-smart-contracts/contracts/LSP8IdentifiableDigitalAsset/LSP8Constants.sol";
+
+contract LSP7Token is LSP7DigitalAsset {
+    constructor(
+        string memory name_,
+        string memory symbol_,
+        address newOwner_,
+        address receiverOfInitialTokens_,
+        uint256 lsp4TokenType_,
+        bool isNonDivisible_,
+        uint256 totalSupply_,
+        bytes memory lsp4MetadataURI_
+    )
+        LSP7DigitalAsset(
+            name_,
+            symbol_,
+            newOwner_,
+            lsp4TokenType_,
+            isNonDivisible_
+        )
+    {
+        // set SupportedStandards:LSP4DigitalAsset
+        LSP4DigitalAssetMetadata._setData(
+            _LSP4_SUPPORTED_STANDARDS_KEY,
+            _LSP4_SUPPORTED_STANDARDS_VALUE
+        );
+
+        // set LSP8ReferenceContract
+        LSP4DigitalAssetMetadata._setData(
+            _LSP8_REFERENCE_CONTRACT,
+            abi.encodePacked(
+                msg.sender,
+                bytes32(uint256(uint160(address(this))))
+            )
+        );
+
+        // set the lsp4MetadataURI
+        _setData(_LSP4_METADATA_KEY, lsp4MetadataURI_);
+
+        // mint all tokens to the receiver of the initial tokens
+        _mint(receiverOfInitialTokens_, totalSupply_, true, "");
+    }
+
+    // override the _setData function so that the LSP8ReferenceContract is not editable
+    function _setData(
+        bytes32 dataKey,
+        bytes memory dataValue
+    ) internal override {
+        require(
+            dataKey != _LSP8_REFERENCE_CONTRACT,
+            "LSP8ReferenceContractNotEditable"
+        );
+        LSP4DigitalAssetMetadata._setData(dataKey, dataValue);
+    }
+}
+
+```
+
+</details>
+
+## Create the LSP8 token
+
+Now that we have the LSP7 token, we can create the LSP8 token.
+
+### Specify what the LSP8 token will need to do
+
+The LSP8 token will be in charge of deploying the LSP7 token and minting the initial supply. When minting the LSP7 tokens, the deployed LSP7 token will be set as a `tokenId` of the LSP8 token.
```

Comment:
Suggestion. Let me know what you think or if it is too much :)

```suggestion
The LSP8 Collection will be in charge of creating new Sub-Collections. It will do this by deploying a new LSP7 Sub-Collection and mint the initial supply of NFTs for the sub-collection. When minting the NFTs (as LSP7 tokens), the address of the deployed LSP7 sub-collection will be set as a `tokenId` within the LSP8 Collection.
```

Reply:
great thanks